File: gabriel_goncalves/src/Coffee.php
<?php

namespace App;

class Coffee
{

    private $isEmpty = true;

    public function refill()
    {
        return $this->isEmpty = false ;
    }

    public function drink()
    {
        return $this->isEmpty = true;
    }

    public function isEmpty()
    {
        if ($this->isEmpty) {
            $this->refill();
            return $this->isEmpty;
        }

        $this->drink();
        return $this->isEmpty;

    }
}

File: castrolol/coffee.lua
local Coffee = {}
Coffee.__index = Coffee

function Coffee.create()
    local coffee = {}
    
    setmetatable(coffee, Coffee)
    coffee.empty = true
    
    return coffee
end

function Coffee:isEmpty()
    return self.empty
end


function Coffee:drink()
    if(self.empty) then
        error("Error: Has no coffee to drink.")
    end
    self.empty = true
end


function Coffee:refill()
    if(not self.empty) then
        error("Error: The coffee isn't empty.")
    end
    self.empty = false
end


return Coffee